plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/projection/DateProjectionFactory.java

```diff
@@ -0,0 +1,202 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.aws.athena.projection;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.spi.TrinoException;
+import io.trino.spi.type.DateType;
+import io.trino.spi.type.TimestampType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.hive.aws.athena.PartitionProjectionProperties.COLUMN_PROJECTION_FORMAT;
+import static io.trino.plugin.hive.aws.athena.PartitionProjectionProperties.COLUMN_PROJECTION_INTERVAL;
+import static io.trino.plugin.hive.aws.athena.PartitionProjectionProperties.COLUMN_PROJECTION_INTERVAL_UNIT;
+import static io.trino.plugin.hive.aws.athena.PartitionProjectionProperties.COLUMN_PROJECTION_RANGE;
+import static io.trino.plugin.hive.aws.athena.PartitionProjectionProperties.getProjectionPropertyRequiredValue;
+import static io.trino.plugin.hive.aws.athena.PartitionProjectionProperties.getProjectionPropertyValue;
+import static io.trino.plugin.hive.aws.athena.projection.Projection.invalidProjectionException;
+import static java.lang.String.format;
+import static java.time.temporal.ChronoUnit.DAYS;
+import static java.time.temporal.ChronoUnit.HOURS;
+import static java.time.temporal.ChronoUnit.MINUTES;
+import static java.time.temporal.ChronoUnit.MONTHS;
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static java.util.Objects.nonNull;
+
+public class DateProjectionFactory
+        implements ProjectionFactory
+{
+    public static final ZoneId UTC_TIME_ZONE_ID = ZoneId.of("UTC");
+
+    // Limited to only DAYS, HOURS, MINUTES, SECONDS as we are not fully sure how everything above day
+    // is implemented in Athena. So we limit it to a subset of interval units which are explicitly clear how to calculate.
+    // Rest will be implemented if this will be required as it would require making compatibility tests
+    // for results received from Athena and verifying if we receive identical with Trino.
+    private static final Set<ChronoUnit> DATE_PROJECTION_INTERVAL_UNITS = ImmutableSet.of(DAYS, HOURS, MINUTES, SECONDS);
+    private static final Pattern DATE_RANGE_BOUND_EXPRESSION_PATTERN = Pattern.compile("\\s*NOW\\s*(([+-])\\s*([0-9]+)\\s*(DAY|HOUR|MINUTE|SECOND)S?\\s*)?");
```

Comment:
do we have test coverage for this kind of pattern?

Reply:
Yes positive and negative:
https://github.com/trinodb/trino/blob/66c868bf8dc7d1de1bd40a69ad33988eb6404ebc/plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java#L901
https://github.com/trinodb/trino/blob/66c868bf8dc7d1de1bd40a69ad33988eb6404ebc/plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java#L1148
https://github.com/trinodb/trino/blob/66c868bf8dc7d1de1bd40a69ad33988eb6404ebc/plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java#L1380